specification/communication/data-plane/Sms/stable/2026-01-23/communicationservicessms.json

```diff
@@ -0,0 +1,613 @@
+      "get": {
+        "tags": [
+          "DeliveryReports"
+        ],
+        "summary": "Gets delivery report for a specific outgoing message.",
+        "operationId": "DeliveryReports_Get",
+        "x-ms-examples": {
+          "GetDeliveryReport": {
+            "$ref": "./examples/DeliveryReportGet.json"
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "outgoingMessageId",
+            "in": "path",
+            "description": "The identifier of the outgoing message.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "api-version",
+            "in": "query",
+            "description": "Version of API to invoke.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeliveryReport"
+            }
+          },
+          "404": {
+            "description": "Not Found - Delivery report for this message is not found",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SmsRecipient": {
+      "description": "Recipient details for sending SMS messages.",
+      "required": [
+        "to"
+      ],
+      "type": "object",
+      "properties": {
+        "to": {
+          "description": "The recipient's phone number in E.164 format.",
+          "type": "string"
+        },
+        "repeatabilityRequestId": {
+          "description": "If specified, the client directs that the request is repeatable; that is, the client can make the request multiple times with the same Repeatability-Request-ID and get back an appropriate response without the server executing the request multiple times. The value of the Repeatability-Request-ID is an opaque string representing a client-generated, 36-character hexadecimal case-insensitive encoding of a UUID (GUID), identifier for the request.",
+          "type": "string"
+        },
+        "repeatabilityFirstSent": {
+          "description": "MUST be sent by clients to specify that a request is repeatable. Repeatability-First-Sent is used to specify the date and time at which the request was first created.eg- Tue, 26 Mar 2019 16:06:51 GMT",
+          "type": "string"
+        }
+      }
+    },
+    "MessagingConnectOptions": {
+      "description": "Defines optional parameters for connecting with the Messaging Connect Partner to deliver SMS.",
+      "required": [
+        "partner",
+        "partnerParams"
+      ],
+      "type": "object",
+      "properties": {
+        "partner": {
+          "description": "Specifies the partner name for message delivery.",
+          "type": "string"
+        },
+        "partnerParams": {
```

Comment:
If partners aren't unbounded, have we considered making this a [polymorphic type](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#polymorphic-types)? This would also translate into strong types on the SDK, and wouldn't force the developer to look up documentation on how to structure their request.
`object` makes most sense when the API shape is controlled directly by the external dependency and the data is just passed through, less so when the API can controlled by the Azure service.

Reply:
@DominikMe If I got it correctly polymorphic types are designed for scenarios where we have a known, finite set of related types that share common behavior but have different implementations. However our `partnerParams` scenario seems different to me as we have an open-ended, extensible key-value structure where each messaging partner can define completely arbitrary parameters.